crates/matrix-sdk-ui/src/encryption_sync/mod.rs

```diff
@@ -43,18 +43,9 @@ use tracing::{debug, trace};
 /// mutex. Using an `EncryptionSync` must then lock that mutex in an owned
 /// way, so that there's at most a single `EncryptionSync` running at any time
 /// in the entire app.
+#[derive(Default)]
 pub struct EncryptionSyncPermit(());
 
-impl EncryptionSyncPermit {
-    /// Create a new [`EncryptionSyncPermit`].
-    ///
-    /// Note: in general, you'd want to get such a permit from a [`SyncService`]
-    /// instead of creating it yourself.
-    pub fn new() -> Self {
```

Comment:
If you make `new` as `pub(crate)`, it's not required to remove it, and to implement `Default` :-p.